scripts/run_e2e_tests.py

```diff
@@ -0,0 +1,472 @@
+    """
+    if not sharding or sharding_instances == 1:
+        return []
+    else:
+        return ['--capabilities.shardTestFiles=%s' % sharding,
+                '--capabilities.maxInstances=%s' % sharding_instances]
+
+
+def get_parameter_for_dev_mode(dev_mode):
+    """Return parameter for whether the test should be running on dev_mode
+
+    Args:
+        dev_mode: bool. Whether the test is running on dev_mode.
+
+    Returns:
+        str: A string for the testing mode command line parameter.
+    """
+    return '--params.devMode=%s' % dev_mode
+
+
+def get_parameter_for_suite(suite):
+    """Return parameter for which suite to be running on the tests.
+
+    Args:
+        suite: str. The suite name to be running on the tests. If the value is
+            `full`, all tests will run.
+
+    Returns:
+        list(str): A list of command line parameter for suite.
+    """
+    return ['--suite', suite]
+
+
+def get_e2e_test_parameters(
+        run_on_browserstack, sharding, sharding_instances, suite, dev_mode):
+    """Return parameters for the end-2-end tests.
+
+    Args:
+        run_on_browserstack: bool. Represents whether the tests should run
+            on browserstack.
+        sharding: bool. Disables/Enables parallelization of protractor tests.
+        sharding_instances: str. Sets the number of parallel browsers to open
+            while sharding.
+        suite: str. Performs test for different suites.
+        dev_mode: bool. Represents whether run the related commands in dev
+            mode.
+    Returns:
+        list(str): Parameters for running the tests.
+    """
+    config_file = get_parameter_for_config_file(run_on_browserstack)
+    sharding_parameters = get_parameter_for_sharding(
+        sharding, sharding_instances)
+    dev_mode_parameters = get_parameter_for_dev_mode(dev_mode)
+    suite_parameter = get_parameter_for_suite(suite)
+
+    commands = [config_file]
+    commands.extend(sharding_parameters)
+    commands.extend(suite_parameter)
+    commands.append(dev_mode_parameters)
+
+    return commands
+
+
+def start_google_engine(dev_mode):
+    """Start the google engine server.
+
+    Args:
+        dev_mode: bool. Represents whether run the related commands in dev
+            mode.
+    """
+    app_yaml_filepath = 'app%s.yaml' % ('_dev' if dev_mode else '')
+
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' % (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_HOME,
+            GOOGLE_APP_ENGINE_PORT, app_yaml_filepath), shell=True)
+    SUBPROCESSES.append(p)
+
+
+def main(args=None):
+    """Run the scripts to start end-to-end tests."""
+
+    parsed_args = _PARSER.parse_args(args=args)
+    other_instance_running = check_running_instance(
+        OPPIA_SERVER_PORT, GOOGLE_APP_ENGINE_PORT)
```

Comment:
I think you can move these into the check_running_instance function if they're going to be the same every time. No need to pass them.

That way, you can actually make that function more oppia-specific and call it is_oppia_server_already_running().

Reply:
Updated